### ISSUE TITLE
Two factor authentication?

### ISSUE THREAD

**Issue by kjartanvalur** [mmacneil/AngularASPNETCore2WebApiAuth#5]
Hi
Thanks for this awesome code sample.
Do you think it´s possible to add two factor authentication via SMS or Email?

Kind regards

**Comment by mmacneil** [comment]
Hi @kjartanvalur - thanks for the feedback.  Re: two factor authentication this demo could be extending very easily to include that.  In fact, you can get the code to plug in by creating a new asp.net mvc web application, include authentication -> "Individual User Accounts" that project template creates a very rich AccountsController for you with 2fa, password reset etc...  I just pasted into this gist: https://gist.github.com/mmacneil/e7e234569802e74e93500bf2abee98dc#file-cs

That should have everything you need to implement 2fa in your demo project.

Regards,
Mark

**Comment by mmacneil** [comment]
Yes, so you don't need to use razor/mvc - that example shows you most of the logic/workflow which you should be able to port (relatively) easily into your restful api to do 2fa.  You should be able to ignore the mvc bits (ie. don't review View()) - see the LoginWith2fa() actions.

Hope this helps.
Mark
